File: C/binarysearch/llama-3.2-90b-vision-preview/llama-3.2-90b-vision-preview.c
#include <stdio.h>
#include <stdlib.h>

/**
 * Binary search function.
 *
 * @param arr   The sorted array to search in.
 * @param target The value to search for.
 * @param low   The starting index of the search range.
 * @param high  The ending index of the search range.
 *
 * @return The index of the target value if found, -1 otherwise.
 */
int binary_search(int* arr, int target, int low, int high) {
    while (low <= high) {
        int mid = low + (high - low) / 2;

        // Check if target is present at the mid
        if (arr[mid] == target)
            return mid;

        // If target is greater, ignore left half
        if (arr[mid] < target)
            low = mid + 1;

        // If target is smaller, ignore right half
        else
            high = mid - 1;
    }

    // If we reach here, then element was not present
    return -1;
}

int main() {
    // Create a sorted array of 1000 values
    int* arr = (int*)malloc(1000 * sizeof(int));
    for (int i = 0; i < 1000; i++) {
        arr[i] = i;
    }

    // Search for a value in the array
    int target = 512;
    int result = binary_search(arr, target, 0, 999);

    // Check if the value was found
    if (result != -1) {
        // printf("Element found at index %d\n", result);
    } else {
        // printf("Element not found in array\n");
    }

    free(arr);
    return 0;
}

File: Python/binarysearch/llama-3.2-90b-vision-preview/llama-3.2-90b-vision-preview.py
def binary_search(arr, target):
    """
    Searches for an element in a sorted array using binary search.

    Args:
        arr (list): A sorted list of elements.
        target: The element to search for.

    Returns:
        int: The index of the target element if found, -1 otherwise.
    """
    low = 0
    high = len(arr) - 1

    while low <= high:
        mid = (low + high) // 2
        if arr[mid] == target:
            return mid
        elif arr[mid] < target:
            low = mid + 1
        else:
            high = mid - 1

    return -1

import random

def main():
    # Generate a sorted list of 1000 integers
    sorted_list = sorted([random.randint(1, 1000) for _ in range(1000)])

    # Perform binary search for a random target
    target = random.choice(sorted_list)
    binary_search(sorted_list, target)

if __name__ == "__main__":
    main()
